src/main/java/keycontacts/logic/commands/CancelLessonCommand.java

```diff
@@ -4,43 +4,103 @@
 
 /**
- * Cancels a currently schedule lesson.
+ * Cancels a currently scheduled lesson.
  */
 public class CancelLessonCommand extends Command {
 
     public static final String COMMAND_WORD = "cancel";
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Cancels the lesson identified by the date and start time.\n"
-            + "Parameters: "
+            + "Parameters: INDEX (must be a positive integer) "
             + PREFIX_DATE + "DATE "
             + PREFIX_START_TIME + "START_TIME\n"
             + "Example: " + COMMAND_WORD + PREFIX_DATE + "06-07-2022 " + PREFIX_START_TIME + "12:00";
-
+    public static final String MESSAGE_SUCCESS = "Cancelled lesson on %1s$ for student: %2s$";
+    public static final String MESSAGE_LESSON_NOT_FOUND = "No lesson with the "
+            + "specified parameters found for student %1s$. "
+            + "Check if you inputted the right timing and student!";
+    private final Date date;
     private final Time startTime;
-    private final LocalDate date;
+    private final Index index;
 
     /**
-     * @param date of the lesson to be cancelled
-     * @param startTime of the lesson to be cancelled
+     * @param index Index of the student attached to the lesson to be cancelled
+     * @param date Date of the lesson to be cancelled
+     * @param startTime Start time of the lesson to be cancelled
      */
-    public CancelLessonCommand(LocalDate date, Time startTime) {
-        requireNonNull(startTime);
+    public CancelLessonCommand(Index index, Date date, Time startTime) {
+        requireNonNull(index);
         requireNonNull(date);
+        requireNonNull(startTime);
 
-        this.startTime = startTime;
+        this.index = index;
         this.date = date;
+        this.startTime = startTime;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return new CommandResult(""); //placeholder
+        requireNonNull(model);
+
+        RegularLesson lessonToCancel = null;
+        List<Student> lastShownList = model.getFilteredStudentList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        Student studentToUpdate = lastShownList.get(index.getZeroBased());
+        boolean matchesLesson = studentToUpdate.matchesLesson(date, startTime);
+
+        if (matchesLesson) {
+            lessonToCancel = studentToUpdate.getRegularLesson();
+        } else {
+            throw new CommandException(String.format(MESSAGE_LESSON_NOT_FOUND, Messages.format(studentToUpdate)));
+        }
+
+        CancelledLesson cancelledLesson = new CancelledLesson(date);
+        Student updatedStudent = studentToUpdate.withAddedCancelledLesson(cancelledLesson);
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS,
+                Messages.format(lessonToCancel), Messages.format(updatedStudent)));
```

Comment:
Also, I realize it may be a bit odd to display the day. Perhaps we should display back the date and time instead?

above:
```
public static final String MESSAGE_SUCCESS = "Cancelled lesson on %1$s at %2$s for student: %3$s";
```

here:
```
return new CommandResult(String.format(MESSAGE_SUCCESS, date, startTime, Messages.format(updatedStudent)));
```